src/infer/inferUserAgent.ts

```diff
@@ -59,6 +59,9 @@ export function getUserAgentString(
     case 'linux':
       userAgent = `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${chromeVersion} Safari/537.36`;
       break;
+    case 'all':
+      userAgent = `Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${chromeVersion} Safari/537.36`;
```

Comment:
> Care to either do more refactor or abandon it?

Sure. I'm not married to it, so whatever's best for the project is fine by me.

> Also, in the first place, why do you need it?

It would seem useful to me if generating an app for downstream distribution, but I'll readily admit that this is not how I use Nativefier, nor do I know what the real world proportion of people who use it that way versus just to create a personal, one off app. As well for me it seemed like an easy win from the perspective of `electron-packager` supported it, so why not us? But perhaps that's the wrong way of looking at it and it's added complexity with no real world benefit for the majority of users. Would all depend on who Nativefier's actual user base is and what their use cases are (of which I readily admit I do not have a good awareness).

> I'll listen, but it seems like a feature that is both problematic and YAGNI anyway, so if it's just for the Docker build I'd like it removed.

The Docker build reason was only a small part of the overall reason, though that was convenient for ensuring that my changes did not break the app build. My main reasoning was that it seemed like a valuable thing to support from the perspective of giving the user a convenient way to build multiple platforms, though admittedly hand rolling a few commands joined by `&&` is not terribly onerous. So if we decide this use case is worth it, great. However, if we decide it's added complexity for very little real world benefit, I could see that too. As the current maintainer of the project, I would naturally follow what you think is best.

Reply:
Oh and:

> Why should all end up as AppleWebkit?

I just grabbed the first entry and removed the OS/Arch and didn't see that part. I'll drop that.

Reply:
Although looking at it here are all the current ones:

```
  darwin: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${DEFAULT_CHROME_VERSION} Safari/537.36`,
  mas: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${DEFAULT_CHROME_VERSION} Safari/537.36`,
  win32: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${DEFAULT_CHROME_VERSION} Safari/537.36`,
  linux: `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${DEFAULT_CHROME_VERSION} Safari/537.36`,
```

All have AppleWebKit, all have Safari. Not sure about either of those.

Reply:
@TheCleric thanks for the exhaustive follow-up and data about `all`. My updated thoughts:

1. The seriously increased Docker size you [measure in your other post](https://github.com/nativefier/nativefier/pull/1122#discussion_r584944697) for niche use case is definitely a concern. A user doing a ton of builds for all platforms would have benefited from your `all` code, but this user is hypothetical to me until I've heard about them, and even then they would remain marginal, and not worth the high bandwidth cost demanded to all users (remember not everybody has a good internet connection).
2. *"That was convenient for ensuring that my changes did not break the app build"*: okay I understand, but doing so in the final for-users Dockerfile doesn't feel like the right place to do so. We can always do with more integration tests, but they should belong to integration tests, not to the final paid-by-users Docker image.
3. On the need for `all`:
    1. As you wrote, a few `&&` chains do the job well
    2. If (and that's a real if) we want something better, maybe we should instead have some kind of `arch1,arch2,arch3` syntax, rather than `all`, whose sole users would be us in a CI job, increasing complexity and command-line API bloat for our sole needs? And even then, not convinced of the benefits of `arch1,arch2,arch3` vs a few `&&`s.
4. You might have twitched on me writing "increasing complexity" above, as this isn't rocket science, and given the quality of what you're pushing, I'm certain you'd easily iron out the bugs (useragent, icon) I pointed with a refactor. My sensitivity here is that for now, Nativefier is in super-minimal maintenance mode (see my blah at the top of README.md), so I'm extremely picky and ruthless about what gets in: anything not a bugfix or not clearly valuable I'd rather leave out.
    - On that, by the way, my message in README is a double-edged sword: I'm encouraging contributions, but on the other side I'm clearly not an "accept all" maintainer. :shrug: , I may try a rephrase but it's difficult, I definitely don't want to discourage, but sometimes submitters get frustrated that I heavily bend their PRs, or alright reject them. More often than not, though, they seem to understand, and things go well, like I think is happening here. I welcome your feedback about it.

So, thanks for mentioning you're not super attached to this feature and would agree to remove it. Given this discussion, yup I confirm I'd like to see it removed from the PR.

Reply:
> so I'm extremely picky and ruthless about what gets in: anything not a bugfix or not clearly valuable I'd rather leave out.

To be clear, I thought you were fair in this PR and I was not frustrated about this at all. Having done my share of open source contributions, I've learned by now that what my idea of a good idea may not necessarily fit with what is good for the overall project. You gave me a chance to justify my reasoning, and in balance of risk vs. reward, you made a decision that you think is best for the project (and I can't say you're wrong). If I was super worried about not throwing this code away I would have created an issue first and talked it over. So no big deal. I'll get it removed. 😄